src/ThreadTestFixture.cpp

```diff
@@ -0,0 +1,82 @@
+///
+/// \author	John Farrier
+///
+/// \copyright Copyright 2015 John Farrier
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+
+#include <celero/ThreadTestFixture.h>
+
+#include <iostream>
+#include <algorithm>
+
+#include <assert.h>
+
+using namespace celero;
+
+ThreadTestFixture::ThreadTestFixture()
+{
+}
+
+ThreadTestFixture::~ThreadTestFixture()
+{
+}
+
+void ThreadTestFixture::startThreads(uint64_t threads, uint64_t calls)
+{
+	uint64_t callsPerThread = calls / threads;
+    for (uint64_t i = 0; i < threads; ++i)
+    {
+		this->threads.push_back(std::thread([this, i, callsPerThread]()
```

Comment:
Have you considered using std::async here instead?  This would a void joining later where you could simply wait on Future completion.  Also reference https://akrzemi1.wordpress.com/2012/11/14/not-using-stdthread/, http://scottmeyers.blogspot.com/2013/03/stdfutures-from-stdasync-arent-special.html